discovery.md

```diff
@@ -691,108 +695,102 @@ Content-Type: application/json
 ]
 ```
 
-The format of the HTTP Response MUST adhere to the following:
+The format of the `200 OK` HTTP Response MUST adhere to the following:
 ```
+200 OK
 Content-Type: application/json
 
 [
-  "id": "{id}"
+  {
+    id: "{id}",
+    epoch: "{epoch}"
+  },
   ...
 ]
 ```
+#### `DELETE /services`
 
-#### `POST /services?import`
-
-This MUST add, or update, the specified Services to the list of Services
-available at this Discovery Endpoint. The Body of the request message MUST
-contain an array of zero or more Service Entries. For convinience to clients,
-the presence of the `id` and `epoch` attributes are OPTIONAL. If `id` is
-present then it MUST be retained as a result of this operation and any existing
-Service with that `id` MUST be replaced. If `epoch` is present then the
-`epoch` value of the resulting Service MUST be larger than the maximum of
-the incoming Service and any existing Service with that same `id`. If either
-attribute is not present then the Discovery Endpoint MUST assign appropriate
-value to the missing attribute.
-
-If the Discovery Endpoint is unable to successully add all of the Services
-in the incoming request then an error MUST be generated and none of the
-specified Services are to be added, or updated, in the Discovery Endpoint.
-
-The Services in the request message MUST be processed in the order in which
-they appear. While any error in the overall process MUST rollback any changes
-made due to the request, the processing of each Service in the request MUST be
-applied as if all previous Services in the request were already completed.
-This means that attribute uniqueness checking, such as on the `name`
-attribute, MUST be done on the new state of all Services as if all previously
-specified Services in the incoming request were already successfully processed.
+This MUST delete all of the Services in the Discovery Endpoint that are
+contained in the request.  If any of those Services cannot be deleted for any
+reason (e.g. an `id` is not given or a Service with the given `id` does not
+exist) then the entire request MUST fail with no effect.  The deletion MAY be a
```

Comment:
I think we may need to define what 'soft delete" means w.r.t. the GET - meaning should a soft deleted one appear in there? I guess not. Can someone resurrect one? If not, I wonder if we should just leave this as an impl detail since if people can't see it but it lives for a while then who cares?

Reply:
> should a soft deleted one appear in there?
I would think so by default but I'd be happy to leave it unspecified (I like making sure things are in user's faces when they might be important).  Otherwise I would probably need to be quite explicit about the optional filter and its behavior.  It feels a little weird since the soft delete behavior is optional.  Happy to do whatever is preferred, I have only a default instinct and a weak opinion.

Reply:
I'd prefer that at the API level "delete == delete and it's no longer seen". Whether it lives on in the server is an impl choice as long as the user doesn't see it any more. It keeps the model simple.

Reply:
As discussed we will remove mention of soft deletes from the specification, at least for now.